lib/managedservicesplatform/runtime/contract/diagnostics.go

```diff
@@ -200,3 +215,152 @@ func (c diagnosticsContract) ConfigureSentry(liblog *log.PostInitCallbacks) bool
 	}
 	return sentryEnabled
 }
+
+// JobExecutionCheckIn checks in to indicate a job execution is starting, and
+// returns a callback that should be called when the job execution is complete.
+// This is only valid for MSP cron jobs.
+//
+// Direct usage example:
+//
+//	func work(ctx context.Context) (err error) {
+//		done, err := c.Diagnostics.JobExecutionCheckIn(ctx)
+//		if err != nil { /* failed to register check-in */ }
+//		defer done(err)
+//
+//		// ... do work
+//	}
+//
+// Note the use of named returns in order to correctly capture the final error.
+// Various contract environment variables MUST be set for Sentry monitor check-ins
+// to be enabled, otherwise this method will only render log entries - required
+// variables are set by MSP infrastructure.
+func (c diagnosticsContract) JobExecutionCheckIn(ctx context.Context) (func(err error), error) {
+	// All values must be set by MSP infrastructure
+	useSentryCronMonitor := c.useSentry() &&
+		c.cronSchedule != nil &&
+		c.cronDeadline != nil
+
+	executionID := inferJobExecutionID(ctx)
+	logger := opentelemetry.TracedLogger(ctx, c.internal.logger).
+		Scoped("execution.checkin").
+		With(
+			log.String("executionID", executionID),
+			log.Bool("useSentryCronMonitor", useSentryCronMonitor),
+			log.Stringp("cronSchedule", c.cronSchedule),
+			log.Durationp("cronDeadline", c.cronDeadline),
+		)
+
+	start := time.Now()
+	logger.Info("job execution starting")
+
+	logCompletion := func(err error) {
+		d := log.Duration("duration", time.Since(start))
+		if err != nil {
+			logger.Error("job execution failed", log.Error(err), d)
+		} else {
+			logger.Info("job execution succeeded", d)
+		}
+	}
+
+	if !useSentryCronMonitor {
+		return logCompletion, nil
+	}
+
+	// Set up Sentry client with some metadata, similar to sourcegraph/log
+	// https://github.com/sourcegraph/log/blob/main/internal/sinkcores/sentrycore/worker.go#L96
+	sentryClient, err := sentry.NewClient(sentry.ClientOptions{
+		Dsn:         *c.sentryDSN,
+		Release:     c.internal.service.Version(),
+		Environment: c.internal.environmentID,
+		Tags: map[string]string{
+			"scope":                    "job.execution.checkin",
+			"resource.service.name":    c.internal.service.Name(),
+			"resource.service.version": c.internal.service.Version(),
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "initialize Sentry client from configured DSN")
+	}
+
+	monitor := fmt.Sprintf("%s-%s", c.internal.service.Name(), c.internal.environmentID)
+	monitorConfig := &sentry.MonitorConfig{
+		Schedule:   sentry.CrontabSchedule(*c.cronSchedule),
+		MaxRuntime: int64(math.Ceil(c.cronDeadline.Minutes())),
+		// Half the deadline is used as a margin to allow for a grace period
+		// for execution to report they are active
+		CheckInMargin: int64(math.Ceil(c.cronDeadline.Minutes() / 2)),
+		// Set in MSP
```

Comment:
```suggestion
		// When running in MSP, the timezone is always set to UTC.
```

Assuming this is the actual intention.

"Set in MSP" sounds like an action to do, not a fact grabbed from a source.